feg/gateway/services/s8_proxy/servicers/s8_proxy.go

```diff
@@ -64,30 +66,35 @@ func NewS8ProxyWithEcho(config *S8ProxyConfig) (*S8Proxy, error) {
 func newS8ProxyImp(cli *gtp.Client, config *S8ProxyConfig) (*S8Proxy, error) {
 	// TODO: validate config
 	s8p := &S8Proxy{
-		config:    config,
-		gtpClient: cli,
+		config:        config,
+		gtpClient:     cli,
+		healthTracker: metrics.NewS8HealthTracker(),
 	}
 	addS8GtpHandlers(s8p)
 	return s8p, nil
 }
 
 func (s *S8Proxy) CreateSession(ctx context.Context, req *protos.CreateSessionRequestPgw) (*protos.CreateSessionResponsePgw, error) {
+	metrics.SessionCreateRequests.Inc()
```

Comment:
If there are existing unit tests for the methods that now produce metrics, you could try to add assertions that the correct counters are incremented.

Reply:
I checked the other proxies and there are no such tests, as far as I can tell. In order to access the metric count we need to introduce the helper function below. This does indeed provide the correct counts. 

```
func getMetricValue(counter prometheus.Counter) float64 {
	// Helper function to access the prometheus counter value
    channel := make(chan prometheus.Metric, 1)
    counter.Collect(channel)
    metric := dto.Metric{}
    _ = (<-channel).Write(&metric)
    return *metric.Counter.Value
}
```